app/features/calculator.py

```diff
@@ -0,0 +1,892 @@
+        result = evaluate_binary_operation(node)
+    elif isinstance(node, ast.UnaryOp):
+        result = evaluate_unary_operation(node)
+    elif isinstance(node, ast.Constant):
+        result = node.value
+    elif isinstance(node, ast.Name):
+        if node.id in ALLOWED_CONSTANTS:
+            result = ALLOWED_CONSTANTS[node.id]
+        else:
+            error_message = f"Constant not allowed or syntax error: {node.id}"
+            raise CalculationError(error_message)
+    elif isinstance(node, ast.Call):
+        result = evaluate_function_call(node)
+    elif isinstance(node, ast.Expression):
+        result = evaluate_node(node.body)
+    else:
+        error_message = "Unsupported node type"
+        raise CalculationError(error_message)
+    return result
+
+
+class Calculator(interactions.Extension):
+    """Calculator extension."""
+
+    def __init__(self, bot: interactions.Client) -> None:
+        self.bot = bot
+        print("Calculator extension loaded")
+
+    @interactions.slash_command(
+        name="calculate",
+        description="Calculate a mathematical expression",
+        options=[
+            interactions.SlashCommandOption(
+                name="expression",
+                description="The mathematical expression to calculate",
+                required=True,
+                type=interactions.OptionType.STRING,
+            ),
+        ],
+    )
+    async def calculate_expression(
+        self,
+        ctx: interactions.SlashContext,
+        expression: str,
+    ) -> None:
+        """Calculate the given mathematical expression."""
+        try:
+            result = evaluate_expression(expression)
+            await ctx.send(f"The result of `{expression}` is: {result}")
+        except CalculationError as calc_error:
+            await ctx.send(f"Calculation error: {calc_error}", ephemeral=True)
+        except (ValueError, TypeError) as specific_error:
+            await ctx.send(f"An error occurred: {specific_error}", ephemeral=True)
+        except UnexpectedCalculationError as unexpected_error:
+            await ctx.send(f"An unexpected error occurred: {unexpected_error}", ephemeral=True)
+
+    @interactions.slash_command(name="calc", description="Calculator functions")
+    async def calc(self, ctx: interactions.SlashContext) -> None:
+        """Provide base command for calculator functions."""
+
+    @calc.subcommand(sub_cmd_name="sqrt", sub_cmd_description="Calculate the square root")
+    @interactions.slash_option(
+        name="number",
+        description="Number to calculate the square root of",
+        required=True,
+        opt_type=interactions.OptionType.STRING,
+    )
+    async def calc_sqrt(self, ctx: interactions.SlashContext, number: str) -> None:
+        """Calculate the square root of a number."""
+        try:
+            result = evaluate_expression(number)
+            await ctx.send(f"The square root of {number} is: {result}")
```

Comment:
The actual expression to calculate the sqrt is missing.
Currently `/calc sqrt 81` results in `81`